fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/NormalizeToSlot.java

```diff
@@ -133,7 +136,10 @@ public static NormalizeToSlotTriplet toTriplet(Expression expression, @Nullable
             }
 
             Alias alias = new Alias(expression, expression.toSql());
-            return new NormalizeToSlotTriplet(expression, alias.toSlot(), alias);
+            SlotReference slot = (SlotReference) alias.toSlot();
+            // BE will create a nullable uint8 column to expand NullLiteral when necessary
+            return new NormalizeToSlotTriplet(expression, expression instanceof NullLiteral ? slot.withDataType(
+                    TinyIntType.INSTANCE) : slot, alias);
```

Comment:
if we just want to process `group by null`, maybe a better way to do it is generating NullLiteral with TinyIntType when we do rewrite.